src/sys/signal.rs

```diff
@@ -733,44 +733,78 @@ mod tests {
+        let flags_no_info = SaFlags::SA_ONSTACK | SaFlags::SA_RESTART;
+        let flags_with_info = flags_no_info | SaFlags::SA_SIGINFO;
+        let mut mask = SigSet::empty();
+        mask.add(SIGUSR1);
+        let handler = SigHandler::Handler(handler_fn);
+
+        let action1 = SigAction::new(handler, flags_with_info, mask);
+        assert_eq!(action1.handler(), handler);
+        assert_eq!(action1.flags(), flags_no_info);
+        assert!(action1.mask().contains(SIGUSR1));
+        assert!(!action1.mask().contains(SIGUSR2));
 
-        let handler_sig = SigHandler::Handler(test_sigaction_handler);
+        let action2 = SigAction::new(handler, flags_no_info, mask);
+        assert_eq!(action2.handler(), handler);
+        assert_eq!(action2.flags(), flags_no_info);
+        assert!(action2.mask().contains(SIGUSR1));
+        assert!(!action2.mask().contains(SIGUSR2));
+    }
+
+    #[test]
+    fn test_sigaction_action() {
+        use libc;
 
+        extern fn handler_fn(_: libc::c_int,
+            _: *mut libc::siginfo_t, _: *mut libc::c_void) {}
         let flags_no_info = SaFlags::SA_ONSTACK | SaFlags::SA_RESTART;
         let flags_with_info = flags_no_info | SaFlags::SA_SIGINFO;
-
         let mut mask = SigSet::empty();
         mask.add(SIGUSR1);
+        let handler = SigHandler::SigAction(handler_fn);
 
-        let action_sig1 = SigAction::new(handler_sig, flags_with_info, mask);
+        let action1 = SigAction::new(handler, flags_with_info, mask);
+        assert_eq!(action1.handler(), handler);
+        assert_eq!(action1.flags(), flags_with_info);
+        assert!(action1.mask().contains(SIGUSR1));
+        assert!(!action1.mask().contains(SIGUSR2));
 
-        assert_eq!(action_sig1.flags(), flags_no_info);
-        assert_eq!(action_sig1.handler(), handler_sig);
-
-        mask = action_sig1.mask();
-        assert!(mask.contains(SIGUSR1));
-        assert!(!mask.contains(SIGUSR2));
+        let action2 = SigAction::new(handler, flags_no_info, mask);
+        assert_eq!(action2.handler(), handler);
+        assert_eq!(action2.flags(), flags_with_info);
+        assert!(action2.mask().contains(SIGUSR1));
+        assert!(!action2.mask().contains(SIGUSR2));
+    }
 
-        let action_sig2 = SigAction::new(handler_sig, flags_no_info, mask);
-        assert_eq!(action_sig2.flags(), flags_no_info);
+    #[test]
+    fn test_sigaction_default() {
+        let flags = SaFlags::SA_ONSTACK | SaFlags::SA_RESTART;
+        let mut mask = SigSet::empty();
+        mask.add(SIGUSR1);
 
-        let handler_act = SigHandler::SigAction(test_sigaction_action);
-        let action_act1 = SigAction::new(handler_act, flags_with_info, mask);
-        assert_eq!(action_act1.handler(), handler_act);
-        let action_act2 = SigAction::new(handler_act, flags_no_info, mask);
-        assert_eq!(action_act2.flags(), flags_with_info);
+        let action = SigAction::new(SigHandler::SigDfl, flags, mask);
```

Comment:
This test used to use the `SaFlags::SA__SIGINFO` flag as well. Was there a reason it was removed?